### PR TITLE
Allow ceph-ansible to be run on a locally built/installed Ceph

### PR DESCRIPTION
**Pull request by danielin917**
Only for redhat_install:

-First install ceph into a directory with CMake
	Possibly add to README for when building ceph: cmake -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib -DWITH_SYSTEMD=ON -DCMAKE_INSTALL_PREFIX:PATH:=/usr <ceph_src> && make DESTDIR=<install_dir> install/strip

-Ceph-ansible copies over the install_dir

-User can use rundep_installer.sh to install any runtime dependencies that ceph needs onto the machine from rundep

### CODE REVIEW

**Comment by leseb** [comment]
@danielin917 can you do indent of 2 instead of 4 please?

**Comment by danielin917** [comment]
@leseb indents in rundep_installer.sh have been fixed

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 1]
```diff
@@ -39,13 +39,66 @@
...
 
+- name: make temp dir
+  file: path=/tmp state=directory
+  when:
+    - ceph_origin == 'local'
+    - use_installer
+
+- name: copy rundep installer
+  synchronize:
+        src: "{{ansible_dir}}/rundep_installer.sh"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: indent

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 2]
```diff
@@ -39,13 +39,66 @@
...
+        dest: "/tmp/rundep_installer.sh"
+  when:
+    - ceph_origin == 'local'
+    - use_installer
+
+- name: copy rundep
+  copy:
+        src: "{{ansible_dir}}/rundep"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: indent

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 3]
```diff
@@ -39,13 +39,66 @@
...
+        dest: "/tmp/rundep"
+  when:
+    - ceph_origin == 'local'
+    - use_installer
+
+- name: install ceph dependencies
+  shell: "/tmp/rundep_installer.sh"
+  args:
+        chdir: "/tmp/"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: indent

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 4]
```diff
@@ -39,13 +39,66 @@
...
+- name: synchronize ceph install
+  synchronize:
+    src: "{{ceph_installation_dir}}/"
+    dest: "/"
+  when:
+    - ceph_origin == 'local'
+
+- name: create user group ceph
+  group:
+        name: 'ceph'
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: indent

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 5]
```diff
@@ -39,13 +39,66 @@
...
+  when:
+    - ceph_origin == 'local'
+
+- name: create user ceph
+  user:
+        name: 'ceph'
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: indent

File: roles/ceph-mon/tasks/deploy_monitors.yml [hunk 6]
```diff
@@ -45,15 +45,15 @@
     mode: "{{ dir_mode }}"
 
 - name: ceph monitor mkfs with keyring (for or after infernalis release)
-  command: ceph-mon --cluster {{ cluster }} --setuser ceph --setgroup ceph --mkfs -i {{ monitor_name }} --fsid {{ fsid }} --keyring /var/lib/ceph/tmp/keyring.mon.{{ monitor_name }}
+  command: ceph-mon --cluster {{ cluster }} --setuser ceph --cluster {{ cluster }} --setgroup ceph --mkfs -i {{ monitor_name }} --fsid {{ fsid }} --keyring /var/lib/ceph/tmp/keyring.mon.{{ monitor_name }}
```
Comment (leseb) [inline roles/ceph-mon/tasks/deploy_monitors.yml]: you have 2:  `--cluster {{ cluster }}`

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 7]
```diff
@@ -39,13 +39,66 @@
   include: redhat_ceph_repository.yml
   when: ceph_origin == 'upstream'
 
+- name: make temp dir
+  file: path=/tmp state=directory
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: The name of the task is a bit confusing, are we trying to create a 'real' tempory directory (ie: `mktemp -d`) or just `/tmp`?

The task should be written like this: 

```
file:
  path: /tmp
  state: directoty
```

**Comment by leseb** [comment]
I think the tasks: `name: copy rundep installer` and `name: install ceph dependencies` can be merged into one by using the script module: http://docs.ansible.com/ansible/script_module.html

Would you mind giving it a try?

**Comment by mattt416** [comment]
Will this functionality be tested on commit via the CI system?

**Comment by leseb** [comment]
@mattt416 nop :/

**Comment by leseb** [comment]
Well the default installation will be not the specific feature :)

**Comment by danielin917** [comment]
I tested it with the new script module and mktemp naming and it works. If there are no other issues I can rebase again into one commit.

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 8]
```diff
@@ -39,13 +39,66 @@
...
 
+- name: make sure /tmp exists
+  file:
+    path: /tmp
+    state: directory
+  when:
+    - ceph_origin == 'local'
+
+- name: use mktemp to create name for rundep
+  shell: "mktemp /tmp/rundep.XXXXXXXX"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: command module should be fine here

Comment (danielin917) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: done

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 9]
```diff
@@ -39,13 +39,67 @@
...
+- name: make sure /tmp exists
+  file:
+    path: /tmp
+    state: directory
+  when:
+    - ceph_origin == 'local'
+    - use_installer
+
+- name: use mktemp to create name for rundep
+  command: "mktemp /tmp/rundep.XXXXXXXX"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: use `mktemp -d` to create a directory, otherwise you will get a file

Comment (danielin917) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: I'm only trying to create one file that will hold the installation list.

Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: Oh now I get it.

**Pull request merged**